### PR TITLE
Feature/rpcs

### PR DESCRIPTION
**Pull request by AlexBHarley**

### CODE REVIEW

**Comment by yasserf** [comment]
Interesting commit name =P

Code looks good so far!

**Comment by AlexBHarley** [comment]
@yasserf I think this can be merged before the rpc cucumber tests, to avoid a PR that is too large

**Comment by yasserf** [comment]
awesome, will review today!

File: src/main/java/io/deepstream/rpc/RpcCallback.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package io.deepstream.rpc;
+
+
+public interface RpcCallback {
+
+    void Call( Object data, RpcResponse response );
```
Comment (yasserf) [inline src/main/java/io/deepstream/rpc/RpcCallback.java]: Can we change this name? To 'rpcRequested' or something

**Comment by yasserf** [comment]
Hey Alex,

Code looks really clean. but there a few things I think we can tackle before merging:

1) I added the provide and make calls to a demo app in deepstreamTest. Current make calls a fake name and I expect 'No RPC found' response, but instead get a timeout. 
2) When I get a timeout it kills my connection
3) Connection times out when nothing happens ( maybe java needs a keep alive? Quite interesting )

If you want we can pair for an hour on this when your free and discuss a few ideas!

**Pull request merged**